### PR TITLE
Pass blocked request resource list from outside. so the resources need to be blocked is configuarable

### PR DESCRIPTION
**Pull request by msk86**
With my pull request to `phantom`: https://github.com/sgentle/phantomjs-node/pull/229

Passing a variable to `onResourceRequested` can be done now.

See this note in `phantom`:

* onResourceRequested takes a function that executes in the scope of phantom which has access to request.abort(), request.changeUrl(url), and request.setHeader(key,value). The second argument is the callback which can execute in the scope of your code, with access to just the requestData. This function can apply extra arguments which can be passed into the first function e.g.

  ```
page.onResourceRequested(
    function(requestData, request, arg1, arg2) { request.abort(); },
    function(requestData) { console.log(requestData.url) },
    arg1, arg2
);
```

So I create a json file which include all resources need to be blocked, then pass it into `onResourceRequest`.

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -175,68 +176,24 @@ server.onPhantomPageCreate = function(req, res) {
...
-        (/googleads.g.doubleclick.net/gi).test(requestData.url) ||
-        (/pagead2.googlesyndication.com/gi).test(requestData.url)) {
-
-        request.abort();
+server.onResourceRequested = function (requestData, request, blockedResources) {
+    for(var i = 0,l = blockedResources.length; i < l; i++) {
+        var regex = new RegExp(blockedResources[i], 'gi');
+        if(regex.text(requestData.url)) {
+            request.abort();
+            requestData.aborted = true;
```
Comment (msk86) [inline lib/server.js]: This requestData is a shared object between `onResourceRequested` and `onResourceRequestedCallback`

File: lib/server.js [hunk 2]
```diff
@@ -175,68 +176,24 @@ server.onPhantomPageCreate = function(req, res) {
...
-        (/youtube.com\/embed/gi).test(requestData.url) ||
-        (/cdn.heapanalytics.com/gi).test(requestData.url) ||
-        (/googleads.g.doubleclick.net/gi).test(requestData.url) ||
-        (/pagead2.googlesyndication.com/gi).test(requestData.url)) {
-
-        request.abort();
+server.onResourceRequested = function (requestData, request, blockedResources) {
+    for(var i = 0,l = blockedResources.length; i < l; i++) {
+        var regex = new RegExp(blockedResources[i], 'gi');
+        if(regex.text(requestData.url)) {
```
Comment (thoop) [inline lib/server.js]: should be regex.test

**Comment by thoop** [comment]
Thanks! This would be nice to get rid of the duplication of that array.

How is requestData.aborted shared between the two?

**Comment by jshado1** [comment]
:+1:

**Comment by msk86** [comment]
You did a trick on the `fn` function to make `request` accessable. The `request` is special and cannot be touched out side `onResourceRequest`, but others are just normal objects. I test on them, they do share between `x` and `cb`.

**Comment by thoop** [comment]
Looks good. Thanks @msk86 !

I also made it so you can pass in `blockedResources` in the server.js file like:

```
var prerender = require('prerender');

var server = prerender({
    blockedResources: [ 'google.com' ]
});

server.start();
```

**Pull request merged**